### PR TITLE
Arakoon unittests

### PR DESCRIPTION
**Pull request by kvanhijf**
Next: Speed up tests

### CODE REVIEW

File: ovs/extensions/db/arakoon/tests/test_arakooninstaller.py [hunk 1]
```diff
@@ -18,174 +18,1363 @@
...
+        base_dir_1 = '{0}/base_dir_internal_fwk'.format(mountpoint_1)
+        base_dir_2 = '{0}/base_dir_internal_fwk_extend'.format(mountpoint_2)
+        base_dir_3 = '{0}/base_dir_internal_fwk_extend_port_range'.format(mountpoint_3)
+        client_1 = SSHClient(endpoint=storagerouter_1.ip)
+        client_2 = SSHClient(endpoint=storagerouter_2.ip)
+        client_3 = SSHClient(endpoint=storagerouter_3.ip)
+
+        for mountpoint in [mountpoint_1, mountpoint_2, mountpoint_3]:
+            if os.path.exists(mountpoint) and mountpoint != '/':
+                shutil.rmtree(mountpoint)
```
Comment (khenderick) [inline ovs/extensions/db/arakoon/tests/test_arakooninstaller.py]: You can remove that, as we don't use the filesystem anymore.

File: ovs/extensions/generic/tests/sshclient_mock.py [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+        if last_part is not None:
+            pointer = pointer['dirs'][last_part]
+        return pointer['dirs'].keys() + pointer['files'].keys()
+
+    @staticmethod
+    def symlink(client, links):
+        """
+        Mocked symlink method
+        """
+        _ = client, links
```
Comment (khenderick) [inline ovs/extensions/generic/tests/sshclient_mock.py:220]: Raise an exception instead

File: ovs/extensions/generic/tests/sshclient_mock.py [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+        if client.ip not in MockedSSHClient._file_system:
+            MockedSSHClient._file_system[client.ip] = {'info': {}, 'dirs': {}, 'files': {}}
+
+        if not filename.startswith('/'):
+            raise ValueError('In unittest mode, the paths must be absolute')
+
+        parts = [part for part in filename.strip('/').split('/') if part]
+        pointer = MockedSSHClient._file_system[client.ip]
+        if isinstance(contents, list) or isinstance(contents, dict):
+            contents = json.dumps(contents, indent=4, sort_keys=True)
```
Comment (khenderick) [inline ovs/extensions/generic/tests/sshclient_mock.py]: I'd implement the default behavior. I guess internally it will just to `str(...)` or maybe raise.

**Review by khenderick — CHANGES_REQUESTED** [review]

**Review by khenderick — APPROVED** [review]

**Comment by kvanhijf** [comment]
openvstorage-2.8.0-dev.1490087129.e83425a
openvstorage-backend-1.8.0-dev.1490087503.3302166

**Pull request merged**